### PR TITLE
feat(order, types): Add Credit Line to order module

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds credit line model
- adds credit line change order
- adds credit line summary total

### CODE REVIEW

File: packages/modules/order/src/services/__tests__/util/actions/credit-line-add.spec.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      "current_order_total": 30,
+      "pending_difference": 30,
+      "difference_sum": 0,
+      "paid_total": 0,
+      "refunded_total": 0,
+      "credit_line_total": 0
+    }
+
+    Upon adding a credit line, the order total and the pending difference will increase making it possible for the merchant
+    to request the customer for a payment for an arbitrary reason, or prepare the order balance sheet to then allow
```
Comment (olivermrbl) [inline packages/modules/order/src/services/__tests__/util/actions/credit-line-add.spec.ts:55]: question: I wouldn't have expected credit lines to be used for additional payments, only refunds–can you expand on this?

Comment (riqwan) [reply packages/modules/order/src/services/__tests__/util/actions/credit-line-add.spec.ts:55]: Good catch, this was incorrectly written - I had changed this in my current branch. Will port over the changes in a bit.

`pending_difference` is what the customer needs to pay to settle the difference. If a credit line has been issued, the `credit_line_total` needs to be subtracted from `pending_difference` to accurately depict what should happen, which is not happening in this branch.

Comment (olivermrbl) [reply packages/modules/order/src/services/__tests__/util/actions/credit-line-add.spec.ts:55]: That sounds right yeah

Comment (olivermrbl) [reply packages/modules/order/src/services/__tests__/util/actions/credit-line-add.spec.ts:55]: Do you want to update this branch or do it in a follow up?

Comment (riqwan) [reply packages/modules/order/src/services/__tests__/util/actions/credit-line-add.spec.ts:55]: Can do either. Making good progress rn, so focusing on not hindering that flow. 😄

File: packages/modules/order/src/models/credit-line.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    columnType: "text",
+    nullable: true,
+  })
+  reference: string | null = null
+
+  @Property({
+    columnType: "text",
+    nullable: true,
+  })
+  reference_id: string | null = null
```
Comment (olivermrbl) [inline packages/modules/order/src/models/credit-line.ts:66]: question: How will these references be used in practise?

**Review by olivermrbl — APPROVED** [review]

File: packages/modules/order/src/models/credit-line.ts (hunk 2 above)
Comment (riqwan) [reply packages/modules/order/src/models/credit-line.ts:66]: Its the resource that will be the enabler of this operation, in this case payment-collection and its id.

We can scope it down to payments or refunds, but that would require creating multiple credit lines, which I can't see a benefit for.

**Pull request merged**